python/lsst/pex/config/config.py

```diff
@@ -391,6 +391,14 @@ def _validateValue(self, value):
             msg = "Value %s is not a valid value" % str(value)
             raise ValueError(msg)
 
+    def collectImports(self, instance):
+        """This function should call the collectImports method on all config
+        objects the field may own, and return the union of all the _imports
+        attributes for of those configclasses. This function should return
+        None if there are no configs are owned by this field.
+        """
+        return None
```

Comment:
I think I'd change the signature of this to take an output `set` to modify - that both sidesteps the "empty `set` vs `None`" question and makes the distinction between `Field.collectImports` and `Config.collectImports` more clear.  I bet it will also clean up some of the implementations a bit.  Together, I think that's enough to outweigh the usual preference for returning instead of updating an output argument.

I also think I'd rename both of these to `_collectImports` - there's strong precedent in `pex.config` for a leading underscore to mean "package private and/or global protected", partly because `Field` and `Config` are already so intertwined, and partly because we want to minimize the set of non-leading-underscore names that could be confused with concrete configuration options.